### PR TITLE
hot plug disk feature is not working with kubletPodsDir changed.

### PR DESCRIPTION
**Pull request by qkfrksvl**
It's related to issue https://github.com/kubevirt/kubevirt/issues/7978.

Hotplug-disk feature gate is not working with kubletPodsDir changed.
image
[CR-kubevirt] 

I changed some source code related to hotplug-disk to print the path of kubeletPodsDir in terms of volumeMounter and I found that it still print the default path of kubeletPodsDir.
image

so I fixed some code for forwarding path of KubeletPodsDir to volumeMounter and hot plug feature is worked.
image

**What this PR does / why we need it**:
Fixing hotplug-disk feature gate.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
 
```release-note
None
```

### CODE REVIEW

File: pkg/virt-handler/hotplug-disk/mount.go [hunk 1]
```diff
@@ -117,11 +117,11 @@ type vmiMountTargetRecord struct {
 }
 
 // NewVolumeMounter creates a new VolumeMounter
-func NewVolumeMounter(mountStateDir string) VolumeMounter {
+func NewVolumeMounter(mountStateDir string, kubeletPodsDir string) VolumeMounter {
 	return &volumeMounter{
 		mountRecords:       make(map[types.UID]*vmiMountTargetRecord),
 		mountStateDir:      mountStateDir,
-		hotplugDiskManager: hotplugdisk.NewHotplugDiskManager(),
+		hotplugDiskManager: hotplugdisk.NewHotplugDiskManager(kubeletPodsDir),
```
Comment (alicefr) [inline pkg/virt-handler/hotplug-disk/mount.go:124]: You could use the already existing function [NewHotplugDiskWithOptions](https://github.com/kubevirt/kubevirt/blob/1a8f08103e48c5f2bb2f5826d118507ce7ec1f0c/pkg/hotplug-disk/hotplug-disk.go#L59)

**Comment by alicefr** [comment]
@qkfrksvl could you please sign the commit?

**Comment by qkfrksvl** [comment]
> @qkfrksvl could you please sign the commit?

Sorry to late:) Done with sign

File: pkg/virt-handler/hotplug-disk/mount.go (hunk 1 above)
Comment (qkfrksvl) [reply pkg/virt-handler/hotplug-disk/mount.go:124]: I think that there is no need to divide into two options 'NewHotplugDiskManager' and 'NewHotplugDiskWithOptions'.
Just need to forward the value 'KubeletPodsDir' of virt-handler... what do yo say ?

Comment (alicefr) [reply pkg/virt-handler/hotplug-disk/mount.go:124]: Well, you are changing the function signature where there is a second function already existing that you could use to achieve exactly the same

**Comment by alicefr** [comment]
I totally understand the fix and we might merge it as it is. 
However, we have a similar problem for [containerdisks](https://github.com/kubevirt/kubevirt/blob/1a8f08103e48c5f2bb2f5826d118507ce7ec1f0c/pkg/container-disk/container-disk.go#L45) and we are passing the value in a different way ([here](https://github.com/kubevirt/kubevirt/blob/1a8f08103e48c5f2bb2f5826d118507ce7ec1f0c/pkg/container-disk/container-disk.go#L126) and [here](https://github.com/kubevirt/kubevirt/blob/1a8f08103e48c5f2bb2f5826d118507ce7ec1f0c/cmd/virt-handler/virt-handler.go#L277)). I'm wondering if we could refactor the code to match both cases and having a cleaner flow

File: pkg/virt-handler/hotplug-disk/mount.go (hunk 1 above)
Comment (alicefr) [reply pkg/virt-handler/hotplug-disk/mount.go:124]: +1 for removing the function and pass the path

**Comment by xpivarc** [comment]
@qkfrksvl Can you have a look at unit test failures?

**Comment by qkfrksvl** [comment]
> 

It seems that 'kubeletPods' argument is missed in NewController function with unit test(vm_test.go) .
image

**Comment by alicefr** [comment]
> /approve /release-note-edit `Fixed hotplug of disks with custom kubelet path `
> 
> @alicefr I think this could be done as a follow-up. WDYT?

Definitely, let's do it later

**Comment by qkfrksvl** [comment]
> 



> /hold
> @qkfrksvl Can you amend the required change?

okay I've done with additional merge

**Comment by xpivarc** [comment]
@alicefr Happy if you follow up here.

**Comment by qkfrksvl** [comment]
Can anyone follow up this PR ? 
It's been stucked..

**Comment by alicefr** [comment]
@xpivarc I think it is ready to be merged
Let's fix the bug and then we can refactor the code.
/lgmt

**Comment by xpivarc** [comment]
/hold
Please ensure we have a follow up

**Comment by xpivarc** [comment]
> @xpivarc I think it is ready to be merged Let's fix the bug and then we can refactor the code later. /lgmt

@alicefr Spell check ;)

**Comment by alicefr** [comment]
@qkfrksvl could you please add the release note (you can simply add 'NONE' there)

**Comment by qkfrksvl** [comment]
> 

Done :)

**Pull request merged**